src/core/Form/Textarea/Textarea.test.tsx

```diff
@@ -0,0 +1,207 @@
+      );
+      expect(queryByText('Summer time')).not.toBeNull();
+      rerender(<Textarea labelText="Winter is coming" />);
+      expect(queryByText('Winter is coming')).not.toBeNull();
+    });
+  });
+
+  describe('labelMode', () => {
+    it('hidden: should hide visually-hidden classname   ', () => {
+      const { getByText } = render(
+        <Textarea labelText="To be hidden" labelMode="hidden" />,
+      );
+      const label = getByText('To be hidden');
+      expect(label).toHaveClass('fi-visually-hidden');
+    });
+  });
+
+  describe('visualPlaceholder', () => {
+    it('should have the given text as attribute', () => {
+      const { getByRole } = render(
+        <Textarea
+          labelText="To be hidden"
+          visualPlaceholder="Enter text here"
+        />,
+      );
+      const textarea = getByRole('textbox') as HTMLTextAreaElement;
+      expect(textarea).toHaveAttribute('placeholder', 'Enter text here');
+    });
+  });
+
+  describe('children', () => {
+    it('should have given children as content for textarea', () => {
+      const { getByRole } = render(
+        <Textarea labelText="label">Text written!</Textarea>,
+      );
+      const textarea = getByRole('textbox') as HTMLTextAreaElement;
+      expect(textarea).toHaveTextContent('Text written!');
+    });
+  });
+
+  describe('hintText', () => {
+    it('should have element and correct classname for it', () => {
+      const { getByText } = render(
+        <Textarea labelText="label" hintText="Example hint text" />,
+      );
+      const hintText = getByText('Example hint text');
+      expect(hintText).toHaveClass('fi-textarea_hintText');
+    });
+  });
+
+  describe('statusText', () => {
+    it('should have element and correct classname for it', () => {
+      const { getByText } = render(
+        <Textarea labelText="label" statusText="EROR EROR" />,
+      );
+      const statusText = getByText('EROR EROR');
+      expect(statusText).toHaveClass('fi-textarea_statusText');
+    });
+  });
+
+  describe('status', () => {
+    it('should have error classname', () => {
+      const { container } = render(
+        <Textarea labelText="label" status="error" statusText="EROR EROR" />,
+      );
+      expect(container.firstChild).toHaveClass('fi-textarea--error');
+    });
+  });
+
+  describe('optionalText', () => {
+    it('should have element and correct classname for it', () => {
+      const { getByText } = render(
+        <Textarea labelText="label" optionalText="Optional" />,
+      );
+      const optionalText = getByText('(Optional)');
+      expect(optionalText).toHaveClass('fi-textarea_optionalText');
+    });
+  });
+
+  describe('resize', () => {
+    it('allows only vertical resizing', () => {
+      const { getByRole } = render(
+        <Textarea labelText="label" resize="vertical" />,
+      );
+      const textarea = getByRole('textbox') as HTMLTextAreaElement;
+      expect(textarea).toHaveStyle('resize: vertical');
```

Comment:
Is it possible that there could be style like this (e.g. problematic future change):
`resize: vertical;
resize: both`
?

Not sure how the matcher works, does it look for applied styles or style definitions...

Reply:
In current way this test checks if it has `resize: vertical` style, but does not check if the style is tried to applied with other value. 

So in this case, it could be wise to add new expects to make sure that only one exists, e.g

```js
 expect(textarea).not.toHaveStyle('resize: both');
// and others

```

Reply:
Maybe just test if it breaks if you override the desired style with another option?

Reply:
Test seems to broke if trying to apply the style like this:

```jsx
<Textarea
  labelText="label"
  resize="vertical"
  style={{ resize: 'both' }}
/>
```

Jest:
```
    - Expected

    - resize: vertical;
    + resize: both;
```

Or did you have some other way in your mind for overriding the style?

Reply:
This or a change in baseStyles, but I guess it does not matter where the override is coming from.